pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep.go

```diff
@@ -90,10 +93,47 @@ func (r *removeNatsOperatorStep) removeNatsOperatorResources(context *service.Ac
 	// set the right eventing name, which went lost after rendering
 	manifest.Manifest = strings.ReplaceAll(manifest.Manifest, natsSubChartPath, eventingNats)
 
+	clientSet, err := kubeClient.Clientset()
+	tracker, err := progress.NewProgressTracker(clientSet, logger, progress.Config{Interval: progressTrackerInterval, Timeout: progressTrackerTimeout})
+	if err != nil {
+		return err
+	}
+
 	logger.Info("Removing nats-operator chart resources")
 	// remove all the existing nats-operator resources, installed via charts
-	_, err = kubeClient.Delete(context.Context, manifest.Manifest, namespace)
-	return err
+	statefulSet, err := getStatefulSet(context, kubeClient, eventingNats)
+	if err != nil {
+		return err
+	}
+
+	m := []byte(manifest.Manifest)
+	unstructs, err := kubernetes.ToUnstructured(m, true)
+	for _, u := range unstructs {
+		// since the old nats-operator was deployed using the k8s deployment
+		// do not delete the nats service if there is a statefulSet deployed
+		if statefulSet != nil && u.GetName() == eventingNats && strings.EqualFold(u.GetKind(), serviceKind) {
+			continue
+		}
+
+		//if resource is watchable, add it to progress tracker
+		watchable, err := progress.NewWatchableResource(u.GetKind())
+		if err == nil { //add only watchable resources to progress tracker
+			tracker.AddResource(watchable, namespace, u.GetName())
+		}
```

Comment:
Print the error if `err != nil` so that later we can know which resources were not watched.

Reply:
yes, you are right, I will print it